DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_routing_over_vlan_device.py

```diff
@@ -0,0 +1,178 @@
+import asyncio
+import pytest
+
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.lib.ip.ip_address import IpAddress
+from dent_os_testbed.lib.ip.ip_route import IpRoute
+from dent_os_testbed.lib.os.sysctl import Sysctl
+from dent_os_testbed.lib.bridge.bridge_vlan import BridgeVlan
+
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_traffic_generator_connect,
+    tgen_utils_dev_groups_from_config,
+    tgen_utils_get_traffic_stats,
+    tgen_utils_setup_streams,
+    tgen_utils_get_loss,
+    tgen_utils_start_traffic,
+    tgen_utils_stop_traffic,
+    tgen_utils_get_swp_info,
+)
+pytestmark = [pytest.mark.suite_functional_lacp,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures('cleanup_tgen', 'cleanup_bonds', 'cleanup_bridges', 'cleanup_sysctl')]
+
+
+async def test_lacp_routing_over_vlan_device(testbed):
+    """
+    Test Name: LACP routing
+    Test Suite: suite_functional_lacp
+    Test Overview: Test lacp routing over vlan device
+    Test Procedure:
+    1. Create bridge and 2 bonds
+    2. Enslave DUT port <==> Ixia port 1 to bond1
+        # DUT port <==>  Ixia port2 to bond 2
+        # DUT port <==> Ixia port 3 to bond 2
+        # DUT port <==>  Ixia port 4 to bond 2
+    3. Enslave bond2 to bridge
+    4. Set link up on all participant ports
+    5. Enable IPv4 forwarding
+    6. Configure ip addresses in each LAG and configure route 10.1.1.0/24 via vlan device
+    7. Setup one stream with DIP 10.1.1.1, and one with 2.2.2.3
+    8. Transmit traffic
+    9. Verify no traffic loss
+    """
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        pytest.skip('The testbed does not have enough dent with tgen connections')
+    device = dent_devices[0]
+    dent = device.host_name
+    tg_ports = tgen_dev.links_dict[dent][0]
+    ports = tgen_dev.links_dict[dent][1]
+    bridge = 'bridge_0'
+    bond_1 = 'bond_1'
+    bond_2 = 'bond_2'
+    vid = 10
+    vlan_device = f'{bridge}.{vid}'
+    ip_forward = 'net.ipv4.ip_forward'
+    lag = 'LAG_1'
+    # 1. Create bridge and 2 bonds
+    out = await IpLink.add(input_data=[{dent: [{
+        'device': bridge,
+        'type': 'bridge',
+        'vlan_filtering': 1,
+        'vlan_default_pvid': 10,
+        }]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to add bridge'
+
+    out = await IpLink.set(input_data=[{dent: [{'device': bridge, 'operstate': 'up'}]}])
+    assert out[0][dent]['rc'] == 0, 'Failed setting bridge to state up'
+
+    out = await BridgeVlan.add(input_data=[{dent: [{'device': bridge, 'vid': vid, 'self': True}]}])
```

Comment:
This is redundant. Your bridge already has vlan 10 because you created the bridge with `'vlan_default_pvid': 10`